shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/job/FinishedCheckJob.java

```diff
@@ -93,6 +99,19 @@ public void execute(final ShardingContext shardingContext) {
         }
     }
     
+    private boolean checkJobStatus(final String jobId) {
+        Map<Integer, JobProgress> jobProgressMap = ruleAlteredJobAPI.getProgress(jobId);
+        final AtomicBoolean flag = new AtomicBoolean(false);
+        for (JobProgress each : jobProgressMap.values()) {
+            if (JobStatus.FINISHED.equals(each.getStatus()) || JobStatus.PREPARING_FAILURE.equals(each.getStatus())
+                    || JobStatus.EXECUTE_INVENTORY_TASK_FAILURE.equals(each.getStatus()) || JobStatus.EXECUTE_INCREMENTAL_TASK_FAILURE.equals(each.getStatus())) {
+                flag.set(true);
+                break;
+            }
+        }
```

Comment:
Looks it could be simplified. All sharding items' progress statuses are `EXECUTE_INCREMENTAL_TASK`, then it could do data consistency check, else not.

Reply:
And `JobProgress each` might be null.

Reply:
- The judgment here should include all jobStatus because it is possible to get an exception before incrementalTask.
- JobProgress has added judgment NULL